File: app/src/main/java/com/niiadotei/storehouse/ui/suppliers/SuppliersActivity.java
package com.niiadotei.storehouse.ui.suppliers;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.niiadotei.storehouse.data.DatabaseHelper;
import com.niiadotei.storehouse.databinding.ActivitySuppliersBinding;

public class SuppliersActivity extends AppCompatActivity {
    RecyclerView recyclerView;

    SuppliersViewModel suppliersViewModel;

    DatabaseHelper databaseHelper;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ActivitySuppliersBinding binding = ActivitySuppliersBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setTitle("Suppliers");

        ActionBar actionBar = getSupportActionBar();
        assert actionBar != null;
        actionBar.setDisplayHomeAsUpEnabled(true);

        recyclerView = binding.suppliersRecyclerView;

        databaseHelper = new DatabaseHelper(getApplicationContext());

        recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
        suppliersViewModel = new SuppliersViewModel(this, databaseHelper.getSuppliersArray());
        recyclerView.setAdapter(suppliersViewModel);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            this.finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/niiadotei/storehouse/ui/suppliers/SuppliersActivity.kt
package com.niiadotei.storehouse.ui.suppliers

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.niiadotei.storehouse.data.DatabaseHelper
import com.niiadotei.storehouse.databinding.ActivitySuppliersBinding

class SuppliersActivity : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val binding = ActivitySuppliersBinding.inflate(layoutInflater)
        setContentView(binding.root)

        title = "Suppliers"

        val actionBar = supportActionBar!!
        actionBar.setDisplayHomeAsUpEnabled(true)

        val recyclerView = binding.suppliersRecyclerView

        val databaseHelper = DatabaseHelper(applicationContext)

        recyclerView.layoutManager = LinearLayoutManager(applicationContext)
        val suppliersViewModel = SuppliersViewModel(databaseHelper.suppliersArray)
        recyclerView.adapter = suppliersViewModel
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}